src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -0,0 +1,224 @@
+package de.tum.in.www1.artemis.service.metis.conversation;
+
+import static javax.validation.Validation.buildDefaultValidatorFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import javax.validation.ConstraintViolationException;
+
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.metis.ConversationParticipant;
+import de.tum.in.www1.artemis.domain.metis.conversation.Channel;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.metis.ConversationParticipantRepository;
+import de.tum.in.www1.artemis.repository.metis.conversation.ChannelRepository;
+import de.tum.in.www1.artemis.service.metis.conversation.errors.ChannelNameDuplicateException;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.metis.conversation.dtos.ChannelDTO;
+import de.tum.in.www1.artemis.web.websocket.dto.metis.MetisCrudAction;
+
+@Service
+public class ChannelService {
+
+    public static final String CHANNEL_ENTITY_NAME = "messages.channel";
+
+    private static final String CHANNEL_NAME_REGEX = "^[a-z0-9-]{1}[a-z0-9-]{0,20}$";
+
+    private final ConversationParticipantRepository conversationParticipantRepository;
+
+    private final ChannelRepository channelRepository;
+
+    private final UserRepository userRepository;
+
+    private final ConversationService conversationService;
+
+    public ChannelService(ConversationParticipantRepository conversationParticipantRepository, ChannelRepository channelRepository, UserRepository userRepository,
+            ConversationService conversationService) {
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.channelRepository = channelRepository;
+        this.userRepository = userRepository;
+        this.conversationService = conversationService;
+    }
+
+    /**
+     * Gets the channel with the given id or throws a not found exception
+     *
+     * @param channelId the id of the channel
+     * @return the channel with the given id
+     */
+    public Channel getChannelOrThrow(Long channelId) {
```

Comment:
Should be in Repository?

Reply:
Done in 60be6e3895a0e582d0d73f1312ef2a66fb4dbc51